### PR TITLE
Feature: Update Book Information (read, favorite)

### PR DESCRIPTION
**Pull request by JanaS87**
- create updateBook function
- add put mapping
- add update-function to controller + service
- add service + controller test
- add Edit Icon and Dialog
- add book svg for read- state
- add heart svg for favorite-state

Note: 

It is intended that only the information as to whether a book has been read and is a favorite can be changed. All other information such as title, genre etc. is determined by the publishers

### CODE REVIEW

File: backend/src/test/java/org/schmaelzle/backend/controller/BookControllerTest.java [hunk 1]
```diff
@@ -127,4 +127,43 @@ void getFavoriteBooks_whenCalled_thenReturnFavoriteBooks() throws Exception {
...
+                        .content("""
+                                {
+                                    "title": "Harry Potter",
+                                    "author": "J.K. Rowling",
+                                    "genre": "Fantasy",
+                                    "publisher": "Carlson",
+                                    "isbn": "3-551-55167-7",
+                                    "favorite": true,
+                                    "read": false,
+                                    "blurb": "Bis zu seinem elften Geburtstag glaubt Harry, er sei ein ganz normaler Junge. Doch dann erfährt er, dass er sich an der Schule für Hexerei und Zauberei einfinden soll – denn er ist ein Zauberer! In Hogwarts stürzt Harry von einem Abenteuer ins nächste und muss gegen Bestien, Mitschüler und Fabelwesen kämpfen. Da ist es gut, dass er schon Freunde gefunden hat, die ihm im Kampf gegen die dunklen Mächte zur Seite stehen."
```
Comment (lostspezi) [inline backend/src/test/java/org/schmaelzle/backend/controller/BookControllerTest.java:151]: If you want to get rid of this, you could use the object mapper here :-) Not neccessary!

**Review by lostspezi — APPROVED** [review]
LGTM!:) Just one info ^-^

**Review by MarcellNF — APPROVED** [review]
LGTM!:) See info above :D

File: backend/src/test/java/org/schmaelzle/backend/controller/BookControllerTest.java (hunk 1 above)
Comment (JanaS87) [reply backend/src/test/java/org/schmaelzle/backend/controller/BookControllerTest.java:151]: Thanks for the suggestion! I will keep that in mind :-)

**Pull request merged**